### PR TITLE
Add support for Nielsen analytics

### PR DESCRIPTION
**Pull request by uniquelygeneric**
In reference to ampproject/amphtml issue #3337.

### CODE REVIEW

**Comment by uniquelygeneric** [comment]
@avimehta, can you help identify what the Travis error is? It says there is an unexpected token in the examples file, but I don't see anything wrong with the formatting.

File: extensions/amp-analytics/0.1/vendors.js [hunk 1]
```diff
@@ -890,6 +890,48 @@ export const ANALYTICS_CONFIG = /** @type {!JSONType} */ ({
...
+      'duration': {
+        'on': 'timer',
+        'timerSpec': {
+          'interval': 10,
+          'maxTimerLength': 86400,
+          'immediate': false,
+        },
+        'request': 'cloudapi',
+      },
+    'transport': {
```
Comment (lannka) [inline extensions/amp-analytics/0.1/vendors.js]: you're missing an closing `}` here

File: extensions/amp-analytics/0.1/vendors.js [hunk 2]
```diff
@@ -890,6 +890,48 @@ export const ANALYTICS_CONFIG = /** @type {!JSONType} */ ({
...
+      'beacon': false,
```
Comment (lannka) [inline extensions/amp-analytics/0.1/vendors.js]: just want to check that your server does not support sendBeacon, which is recommended.

File: extensions/amp-analytics/0.1/vendors.js [hunk 3]
```diff
@@ -890,6 +890,48 @@ export const ANALYTICS_CONFIG = /** @type {!JSONType} */ ({
...
+      'cloudapi': 'https://cloudapi.imrworldwide.com/nmapi/v2/${apid}/${clientId(imrworldwide)}/a?b=%7B%22devInfo%22%3A%7B%22devId%22%3A%22${clientId(imrworldwide)}%22%2C%22apn%22%3A%22${apn}%22%2C%22apv%22%3A%22${apv}%22%2C%22apid%22%3A%22${apid}%22%7D%2C%22metadata%22%3A%7B%22static%22%3A%7B%22type%22%3A%22static%22%2C%22section%22%3A%22${section}%22%2C%22assetid%22%3A%22${pageViewId}%22%2C%22segA%22%3A%22${segA}%22%2C%22segB%22%3A%22${segB}%22%2C%22segC%22%3A%22${segC}%22%2C%22adModel%22%3A%220%22%2C%22dataSrc%22%3A%22cms%22%7D%2C%22content%22%3A%7B%7D%2C%22ad%22%3A%7B%7D%7D%2C%22event%22%3A%22playhead%22%2C%22position%22%3A%22${timestamp}%22%2C%22data%22%3A%7B%22hidden%22%3A%22${backgroundState}%22%2C%22blur%22%3A%22${backgroundState}%22%2C%22position%22%3A%22${timestamp}%22%7D%2C%22type%22%3A%22static%22%2C%22utc%22%3A%22${timestamp}%22%2C%22index%22%3A%22${requestCount}%22%7D',
+    },
+    'triggers': {
+      'init': {
+        'on': 'visible',
+        'request': 'session',
+      },
+      'visible': {
```
Comment (lannka) [inline extensions/amp-analytics/0.1/vendors.js]: you can merge this with `init`:
```
'request': ['session', 'cloudapi'],
```

File: extensions/amp-analytics/0.1/vendors.js (hunk 1 above)
Comment (uniquelygeneric) [inline extensions/amp-analytics/0.1/vendors.js]: Thanks for catching this!

File: extensions/amp-analytics/0.1/vendors.js (hunk 2 above)
Comment (uniquelygeneric) [inline extensions/amp-analytics/0.1/vendors.js]: Yes, this was on purpose. Our server only accepts GET requests.

File: extensions/amp-analytics/0.1/vendors.js (hunk 3 above)
Comment (uniquelygeneric) [inline extensions/amp-analytics/0.1/vendors.js]: added

File: extensions/amp-analytics/0.1/vendors.js [hunk 4]
```diff
@@ -890,6 +890,45 @@ export const ANALYTICS_CONFIG = /** @type {!JSONType} */ ({
...
+    'vars': {
+      'apid': '',
+      'section': '',
+      'apv': '1.0',
+      'segA': 'NA',
+      'segB': 'NA',
+      'segC': 'NA',
+    },
+    'requests': {
+      'session': 'https://uaid-linkage.imrworldwide.com/cgi-bin/gn?prd=session&c13=asid,P${apid}&sessionId=${clientId(imrworldwide)},&pingtype=4&enc=false&c61=createtm,${timestamp}&rnd=${random}',
```
Comment (lannka) [inline extensions/amp-analytics/0.1/vendors.js]: for readability, can you also extract `clientId` to a var?

File: extensions/amp-analytics/0.1/vendors.js [hunk 5]
```diff
@@ -890,6 +890,45 @@ export const ANALYTICS_CONFIG = /** @type {!JSONType} */ ({
...
+      'cloudapi': 'https://cloudapi.imrworldwide.com/nmapi/v2/${apid}/${clientId(imrworldwide)}/a?b=%7B%22devInfo%22%3A%7B%22devId%22%3A%22${clientId(imrworldwide)}%22%2C%22apn%22%3A%22${apn}%22%2C%22apv%22%3A%22${apv}%22%2C%22apid%22%3A%22${apid}%22%7D%2C%22metadata%22%3A%7B%22static%22%3A%7B%22type%22%3A%22static%22%2C%22section%22%3A%22${section}%22%2C%22assetid%22%3A%22${pageViewId}%22%2C%22segA%22%3A%22${segA}%22%2C%22segB%22%3A%22${segB}%22%2C%22segC%22%3A%22${segC}%22%2C%22adModel%22%3A%220%22%2C%22dataSrc%22%3A%22cms%22%7D%2C%22content%22%3A%7B%7D%2C%22ad%22%3A%7B%7D%7D%2C%22event%22%3A%22playhead%22%2C%22position%22%3A%22${timestamp}%22%2C%22data%22%3A%7B%22hidden%22%3A%22${backgroundState}%22%2C%22blur%22%3A%22${backgroundState}%22%2C%22position%22%3A%22${timestamp}%22%7D%2C%22type%22%3A%22static%22%2C%22utc%22%3A%22${timestamp}%22%2C%22index%22%3A%22${requestCount}%22%7D',
+    },
+    'triggers': {
+      'init': {
```
Comment (lannka) [inline extensions/amp-analytics/0.1/vendors.js]: `init` -> `visible`

File: extensions/amp-analytics/0.1/vendors.js [hunk 6]
```diff
@@ -890,6 +890,45 @@ export const ANALYTICS_CONFIG = /** @type {!JSONType} */ ({
     },
   },
 
+  'nielsen': {
+    'vars': {
+      'apid': '',
```
Comment (lannka) [inline extensions/amp-analytics/0.1/vendors.js]: no need to specify. it's by default empty string

**Comment by lannka** [comment]
please fix the travis build. and also follow the googlebot to sign CLA

**Comment by uniquelygeneric** [comment]
@googlebot, CLA signed for Nielsen

**Comment by lannka** [comment]
> Once you've signed, please reply here (e.g. I signed it!) and we'll verify.

Try this.

**Comment by uniquelygeneric** [comment]
I signed it!

**Comment by lannka** [comment]
@uniquelygeneric 
> If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address. Check your existing CLA data and verify that your email is set on your git commits.
> If you signed the CLA as a corporation, please let us know the company's name.

**Comment by uniquelygeneric** [comment]
@lannka this was signed under kale.smith@nielsen.com on Nielsen's behalf.

**Comment by lannka** [comment]
It should be either the code author or the corp sign it. Can you either sign it yourself or ask Kale to sign it using corp's name?

**Comment by uniquelygeneric** [comment]
@lannka I am Kale and I signed the contract with Nielsen's name...I'm not sure why it's not getting picked up by the bot

**Comment by bpaduch** [comment]
I will investigate with CLA on this.

**Comment by uniquelygeneric** [comment]
@bpaduch The CLA should have been signed by the appropriate party now. Thanks!

**Comment by jooj123** [comment]
Can we get this one merged?

**Comment by uniquelygeneric** [comment]
Yes, we've received the signed copy of the CLA, so I don't see anything holding this back anymore.

@bpaduch, is there anything I can do to help push this along?

**Comment by bpaduch** [comment]
@uniquelygeneric - I'm waiting on response from the CLA team on status before proceeding.  As soon as I get details, I'll advise.

**Comment by bpaduch** [comment]
Hi @uniquelygeneric, the CLA for Nielsen is registered under the following group: `googleclientservicecoreteam@nielsen.com`. Please make sure your email is a member of that group.  Check with the manager of that group (Cicely) to verify you're part of it.  If the CLA still doesn't work, I'll escalate further.

**Comment by uniquelygeneric** [comment]
@bpaduch I have been added to that list, but it could take 24-48 hours for your systems to update. Is there anything that can be done to ensure this doesn't delay the merge by another week?

**Comment by jridgewell** [comment]
@erwinmombay: Can you force merge this?

**Pull request merged**